### PR TITLE
mindustry

### PR DESCRIPTION
**Pull request by EmmmM9O**
for mindustry

### CODE REVIEW

File: types/mindustry/tslint.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{ 
+    "extends": "@definitelytyped/dtslint/dt.json",
+    "rules":{
+	"no-unnecessary-class":false,
```
Comment (DangerBotOSS) [inline types/mindustry/tslint.json:4]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by EmmmM9O** [comment]
publish

File: types/mindustry/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,294 @@
+// Type definitions for mindustry 0.1
+// Project: https://github.com/squi2rel/node-mindustry
+// Definitions by: EmmmPad <https://github.com/EmmmM9O>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.3
+/// <reference types="node" />
```
Comment (peterblazejewicz) [inline types/mindustry/index.d.ts:6]:
```suggestion

/// <reference types="node" />

```

File: types/mindustry/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+import { Socket } from 'net';
+import * as dgram from 'dgram';
+import { EventEmitter } from 'stream';
+export  const debug: boolean;
```
Comment (peterblazejewicz) [inline types/mindustry/index.d.ts:10]:
```suggestion

export  const debug: boolean;
```

File: types/mindustry/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+ export class DataStream {
+    #pos: number;
+    #buf: Buffer;
+    #lim: number;
```
Comment (peterblazejewicz) [inline types/mindustry/index.d.ts:14]: are those private?
if yes, how those are supposed to be used?

File: types/mindustry/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+    put(bytes: any): DataStream;
+    hasRemaining(): boolean;
+    capacity(): number;
+    putShort(data: number): DataStream;
+    putUShort(data: number): DataStream;
+    array(): number[];
+    putInt(data: number): DataStream;
+    getInt(): number;
+    toString(): string;
+    _getBuffer(offset: number): Buffer;
```
Comment (peterblazejewicz) [inline types/mindustry/index.d.ts:37]: is this one private by convention? (wihtout #)

File: types/mindustry/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+ export class TypeIO {
+    static writeString(buf: DataStream, string: string): void;
+    static readString(buf: DataStream): string | null;
+    static writeKick<T extends { id: number }>(buf: DataStream, reasson: T): void;
+    static readKick(buf: DataStream): KickReason;
+    static readStrings(buf: DataStream): string[][];
+}
+
+ export class StreamBegin extends Packet {
+    _id_: number;
```
Comment (peterblazejewicz) [inline types/mindustry/index.d.ts:87]: there seems to be lot of `_id`, are those private and supposed to be exported?

File: types/mindustry/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+    sendChatMessage(msg: string): void;
+    handleClientReceived(packet: Packet): void;
+    loadWorld(packet: unknown, game: unknown): void;
+}
+export function pingHost(port: string, ip: string, callback: (p: unknown, err?: Error) => void): void;
+ export class Mindustry {
+    netClient: NetClient;
+    constructor();
+}
+// Definitions by: EmmmM9O <https://github.com/EmmmM9O>
```
Comment (peterblazejewicz) [inline types/mindustry/index.d.ts:293]: ???
can you remove thos comments, unless part of the header (so move them to the header)
Please also run formatter on your code:
https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes

**Review by peterblazejewicz — COMMENTED** [review]
@EmmmM9O please review comments, thx!

**Review by jakebailey — CHANGES_REQUESTED** [review]
See @peterblazejewicz's comments.

**Pull request closed without merging**